spark-plugins/src/main/java/co/cask/hydrator/plugin/spark/SparkUtils.java

```diff
@@ -84,7 +85,7 @@ static void validateConfigParameters(Schema inputSchema, @Nullable String featur
       throw new IllegalArgumentException("Cannot specify values for both featuresToInclude and featuresToExclude. " +
                                            "Please specify fields for one.");
     }
-    Map<String, Integer> fields = new HashMap<>();
+    Map<String, Integer> fields = new LinkedHashMap<>();
```

Comment:
We are using it in the `DecisionTreePredictor#transform()` and `DecisionTreeTrainer#run()` to create the sparse vector of the features to be used for training/prediction.For every structured record the [featureList ](https://github.com/caskdata/hydrator-plugins/blob/5057d5af2ab52ec2f5fef432df6fc8f01d04cebb/spark-plugins/src/main/java/co/cask/hydrator/plugin/spark/DecisionTreeTrainer.java#L90) and [featureIndex ](https://github.com/caskdata/hydrator-plugins/blob/5057d5af2ab52ec2f5fef432df6fc8f01d04cebb/spark-plugins/src/main/java/co/cask/hydrator/plugin/spark/DecisionTreeTrainer.java#L91) are created for the fields in the sequence retrieved [getFeatureList().keySet()](https://github.com/caskdata/hydrator-plugins/blob/5057d5af2ab52ec2f5fef432df6fc8f01d04cebb/spark-plugins/src/main/java/co/cask/hydrator/plugin/spark/DecisionTreeTrainer.java#L79).Random order returned from `SparkUtils#getFeatureList()` will result in incorrect sequence of values in the sparse vector in case of multiple reducers.
Please let me know your thoughts.